pkg/server/methods/executeCommand.go

```diff
@@ -24,36 +33,39 @@ func (methods *Methods) ExecuteCommand(reply jsonrpc2.Replier, req jsonrpc2.Requ
 	case "setToken":
 		param, ok := arguments[0].(string)
 		if !ok {
-			return reply(methods.Ctx, nil, nil)
+			return reply(methods.Ctx, nil, BadArgumentError{"token"})
 		}
 		methods.setToken(param)
 		methods.updateAllCachedFiles()
 
 	case "setSelfHostedUrl":
 		param, ok := arguments[0].(string)
 		if !ok {
-			return reply(methods.Ctx, nil, nil)
+			return reply(methods.Ctx, nil, BadArgumentError{"selfHostedURL"})
 		}
 		methods.setHostUrl(param)
 		methods.updateAllCachedFiles()
 
 	case "setUserId":
 		param, ok := arguments[0].(string)
 		if !ok {
-			return reply(methods.Ctx, nil, nil)
+			return reply(methods.Ctx, nil, BadArgumentError{"userId"})
 		}
 		methods.setUserId(param)
 
 	case "getWorkflows":
 		content, okContent := arguments[0].(string)
+		if !okContent {
+			return reply(methods.Ctx, nil, BadArgumentError{"fileContent"})
+		}
 		fileUri, okUri := arguments[1].(string)
-		if !okContent || !okUri {
-			return reply(methods.Ctx, nil, nil)
+		if !okUri {
+			return reply(methods.Ctx, nil, BadArgumentError{"fileURI"})
 		}
 
 		parsedFile, err := parser.ParseFromContent([]byte(content), methods.LsContext, uri.File(fileUri), protocol.Position{})
 		if err != nil {
-			return reply(methods.Ctx, nil, nil)
+			return reply(methods.Ctx, nil, errors.New("unable to parse file"))
```

Comment:
Likewise, here, can we re-use existing errors from the protocol?

We have [`InternalError`](https://pkg.go.dev/go.lsp.dev/jsonrpc2#Code) code inside the `jsonrpc2` package. The LSP specification also introduced a [`RequestFailed`](https://github.com/microsoft/language-server-protocol/blob/6d308a755cddef48e8a03c150e871f4a6c8de94f/_specifications/lsp/3.18/specification.md?plain=1#L275) code in v3.17.0, but the Go package we use [only supports v3.15.3](https://github.com/go-language-server/protocol/blob/v0.12.0/version.go#L7).